master/security/icmp-ping.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Use ICMP/ping in policy rules
+---
+
+### Big picture
+
+Use Calico network policy to allow and deny ICMP/ping messages. 
+
+### Value
+
+The **Internet Control Message Protocol (ICMP)** provides valuable network diagnostic functions, but it can also be used maliciously. Attackers can use it to learn about your network, or for DoS attacks. Using Calico network policy, you can control where ICMP is used. For example, you can:
+
+- Allow ICMP ping, but only within your cluster 
+- Allow ICMP for specifically labeled pods   
+  For example, allow ICMP for pods launched by operators for diagnostic purposes, but block other uses.
+- Temporarily enable ICMP to diagnose a problem, then disable it after the problem is resolved
+- Deny/allow ICMP messages (ICMPv4 and/or ICMPv6)
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**GlobalNetworkPolicy** or **NetworkPolicy** with ICMPv4 and ICMPv6 and positive/negative match criteria.
+
+
+### Concepts
+
+#### ICMP packet type and code
+
+Calico network policy also lets you deny and allow specific parts of the ICMP packet for fine-grain control. For example, you can specify ICMP type 5, code 2. For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
+
+### How to
+
+- [Deny all ICMP](#deny-all-icmp)
+- [Allow ICMP ping only within a cluster](#allow-icmp-ping-only-within-a-cluster)
+- [Allow ICMP matching protocol type and code](#allow-icmp-matching-protocol-type-and-code)
+
+#### Deny all ICMP
+
+In this example, all pods are blocked from sending or receiving **ICMPv4** and **ICMPv6** messages. If **ICMPv6** messages are not used in your deployment, it is still good practice to deny them specifically as shown below. 
```

Comment:
We don't explicitly call out that applying this policy will cause a global default-deny behavior as a side-effect.

Maybe we don't need to, since we're assuming users have read the intro to network policy doc, but it does mean that applying this policy without any other policies will block _all traffic_ to and from any workload in the cluster.

Feels like we might want to mention that explicitly. WDYT?

Reply:
I agree. Good catch. I suspect we should add a statement wherever we introduce a "deny all" examples. See if text I added works for you.